src/net/sourceforge/kolmafia/persistence/ConsumablesDatabase.java

```diff
@@ -924,120 +833,90 @@ public static double getAdventureRange(final String name) {
                   false,
                   false)
               .get(name);
-    } else if (ConsumablesDatabase.getRawSpleenHit(name) != null) {
+    } else if (consumable.getRawSpleenHit() != null) {
       range = ConsumablesDatabase.getAdventureMap(perUnit, false, false, false, false).get(name);
     }
 
     if (range == null) {
       return 0.0;
     }
 
-    range += ConsumablesDatabase.conditionalExtraAdventures(name, perUnit);
+    range += ConsumablesDatabase.conditionalExtraAdventures(consumable, perUnit);
 
     return range;
   }
 
-  private static int getStatUnit(final String name) {
+  private static int getStatUnit(Consumable consumable) {
     if (!Preferences.getBoolean("showGainsPerUnit")) {
       return 1;
     }
-    int unit = 0;
-    Integer fullness = ConsumablesDatabase.getRawFullness(name);
-    Integer inebriety = ConsumablesDatabase.getRawInebriety(name);
-    Integer spleenhit = ConsumablesDatabase.getRawSpleenHit(name);
-
-    if (fullness != null) {
-      unit += fullness;
-    }
-    if (inebriety != null) {
-      unit += inebriety;
-    }
-    if (spleenhit != null) {
-      unit += spleenhit;
-    }
+    int unit = consumable.getFullness() + consumable.getInebriety() + consumable.getSpleenHit();
     if (unit == 0) {
       unit = 1;
     }
     return unit;
   }
 
-  public static final String getMuscleByName(final String name) {
-    if (name == null) {
+  public static final String getBaseStatByName(final int stat, final String name) {
+    Consumable consumable = ConsumablesDatabase.consumableByName.get(name);
+    if (consumable == null) {
       return "";
     }
 
-    String range = ConsumablesDatabase.muscleByName.get(name);
-    return range == null ? "" : range;
+    return consumable.statRangeStrings[stat];
   }
 
-  public static final String getMuscleRange(final String name) {
-    if (name == null) {
+  public static final String getStatRange(final int stat, final String name) {
+    return getStatRange(stat, ConsumablesDatabase.consumableByName.get(name));
+  }
+
+  public static final String getStatRange(final int stat, final Consumable consumable) {
+    if (consumable == null) {
       return "+0.0";
     }
 
-    String muscle = ConsumablesDatabase.muscleByName.get(name);
-    double muscleFactor =
-        (KoLCharacter.currentNumericModifier(Modifiers.MUS_EXPERIENCE_PCT) + 100.0) / 100.0;
-    muscleFactor *= ConsumablesDatabase.conditionalStatMultiplier(name);
-    int statUnit = ConsumablesDatabase.getStatUnit(name);
-    int statBonus = ConsumablesDatabase.conditionalExtraStats(name);
-    String range = ConsumablesDatabase.extractStatRange(muscle, muscleFactor, statUnit, statBonus);
+    String statRange = consumable.statRangeStrings[stat];
+    int modifier =
+        switch (stat) {
+          case Consumable.MUSCLE -> Modifiers.MUS_EXPERIENCE_PCT;
+          case Consumable.MYSTICALITY -> Modifiers.MYS_EXPERIENCE_PCT;
+          default -> Modifiers.MOX_EXPERIENCE_PCT;
```

Comment:
Would rather this failed(? I know it can't) a little safer

Reply:
Fixed.